File: design-pattern-jvm/app/src/main/java/design/pattern/java/patterns/behavioral/mediator/Airship.java
package design.pattern.java.patterns.behavioral.mediator;

public class Airship {
  private Mediator controlTower;

  public Airship(Mediator controlTower) {
    this.controlTower = controlTower;
  }

  public void notifyBeforeLand() {
    this.controlTower.notify(this, "beforeLand");
  }

  public void notifyAfterLand() {
    this.controlTower.notify(this, "afterLand");
  }

  public void notifyBeforeLiftOff() {
    this.controlTower.notify(this, "beforeLiftOff");
  }

  public void notifyAfterLiftOff() {
    this.controlTower.notify(this, "afterLiftOff");
  }
}

File: design-pattern-jvm/app/src/main/kotlin/design/pattern/kotlin/patterns/behavioral/mediator/Airship.kt
package design.pattern.kotlin.patterns.behavioral.mediator

class Airship(private val controlTower: Mediator) {
    fun notifyBeforeLand() {
        controlTower.notify(this, "beforeLand")
    }

    fun notifyAfterLand() {
        controlTower.notify(this, "afterLand")
    }

    fun notifyBeforeLiftOff() {
        controlTower.notify(this, "beforeLiftOff")
    }

    fun notifyAfterLiftOff() {
        controlTower.notify(this, "afterLiftOff")
    }
}